SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -173,13 +173,16 @@ void FSpatialObjectRepState::UpdateRefToRepStateMap(FObjectToRepStateMap& RepSta
 	{
 		if (!LocalReferencedObj.Contains(Ref))
 		{
-			TSet<FChannelObjectPair>& RepStatesWithRef = RepStateMap.FindChecked(Ref);
+			TSet<FChannelObjectPair>* RepStatesWithRef = RepStateMap.Find(Ref);
 
-			RepStatesWithRef.Remove(ThisObj);
-
-			if (RepStatesWithRef.Num() == 0)
+			if (ensure(RepStatesWithRef))
```

Comment:
Is there an assumption that this can actually fail in normal usage?

Reply:
It should not, modulo a mistake, like the one fixed by https://github.com/spatialos/UnrealGDK/pull/1694
If there are any more mistakes to unearth, a test build will fire the assertion, but at least it won't take down a release build.